systemvm: list systemvm does not return agent state and version

rhtyd opened the pull request:
This makes the listSystemVms API to return the host status (agent state),
version and last pinged information. This makes it possible for UIs
to call a single API to get this information.

This is largely for Primate.

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos6 ✔centos7 ✔debian. JID-777

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

rhtyd commented:
@shwstppr @Pearl1594 please review as well.

Pearl1594 commented:
LGTM - agentstate, lastpinged and version fields seem to be added to the list systemvms response

shwstppr commented on api/src/main/java/org/apache/cloudstack/api/response/SystemVmResponse.java:
```diff
@@ -138,6 +138,10 @@
     private String state;
 
+    @SerializedName("agentstate")
+    @Param(description = "the agent state of the system VM")
```
can we add `since` key here as these are new response params

rhtyd commented:
@DaanHoogland the technical preview targets 4.14, without this fix the implementation in Primate would be complicated.

weizhouapache commented:
code LGTM

@rhtyd can you make some changes on 4.13/4.14 UI ? the agent state is determined  in a complicated way.

rhtyd commented:
That's not very critical, but let me check if it's a quick one. I originally wanted to limit the scope just to apis, wrt Primate.

DaanHoogland commented:
ok, @rhtyd I'm fine with it. merge at will.

rhtyd commented:
@shwstppr thanks, fixed the `since` in params response keys.
@weizhouapache I've fixed the UI as well. Can you review and test the API/UI changes? Thanks.

rhtyd commented:
Thanks @DaanHoogland will merge after smoketests and some UI confirmation (cc @weizhouapache)
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos6 ✔centos7 ✔debian. JID-784

DaanHoogland commented:
@weizhouapache can you confirm the ui?

weizhouapache commented:
@DaanHoogland ui works.

However, lastping returns wrong date/time. we have same issue on other api response. we can fix them together afterwards.

DaanHoogland commented:
sounds like a ugly bug @weizhouapache shift by ten bits != /1000. I think we should move away from the bit shifting when dealing with milliseconds.
Would you consider this not to be a blocker for this PR?

DaanHoogland commented:
@blueorangutan test

blueorangutan commented:
@DaanHoogland a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

rhtyd commented:
@weizhouapache cc @DaanHoogland I used the same logic as here: https://github.com/apache/cloudstack/blob/master/server/src/main/java/com/cloud/api/query/dao/HostJoinDaoImpl.java#L105

weizhouapache commented:
@DaanHoogland what I said is not a blocker. I am ok with this PR.

@rhtyd yes, it has same issue. In response of listhosts and listsystemvms, both shows the last ping time is "1970-01-XXXX" instead of "2020-02-XXXX". we can fix it in the future, not today.
